src/Term/TermFallback.php

```diff
@@ -46,6 +47,28 @@ public function __construct( $requestedLanguageCode, $text, $actualLanguageCode,
 			throw new InvalidArgumentException( '$sourceLanguageCode must be a non-empty string or null' );
 		}
 
+		if ( $sourceLanguageCode === null && $actualLanguageCode === $requestedLanguageCode ) {
```

Comment:
One of the later two `if`s is redundant and can be removed. What about these checks and messages:
1. "$actual must be different from $requested, if no transliteration took place; use Term without fallback instead"
2. "$source must either be null, if no transliteration took place, or different from $requested"